### PR TITLE
Reduce flakiness of test_messaging.py

### PR DESCRIPTION
**Pull request by jamessynge**
Use two Event objects to coordinate the usage of the messaging
forwarder. In particular, once the forwarder has created the sockets
we can create the publisher and subscriber. And when shutting down,
we should wait until the sockets have been closed else we may find
that we can't create the ports for the next test. This means we can
remove the sleep() calls which slow the tests down without providing
confidence.

Unlike PR #439, this uses an instance of multiprocessing.SyncManager,
which means that the Event objects are accessed via proxies
communicating with a server process. This avoids the problem that
[TravisCI doesn't directly support shared memory](https://github.com/travis-ci/travis-cookbooks/issues/155) without funny
workarounds. This avoids the issue entirely.

Note that this reduces the flakiness on my laptop from perhaps
3% to around 0.2%, which is an improvement, but there is still
room to go.

Partial fix for #131.

### CODE REVIEW

File: pocs/tests/test_messaging.py [hunk 1]
```diff
@@ -1,21 +1,46 @@
+import multiprocessing
 import pytest
 import time
 
 from datetime import datetime
-from multiprocessing import Process
 from pocs.utils.messaging import PanMessaging
 
 
+@pytest.fixture(scope='module')
```
Comment (wtgee) [inline pocs/tests/test_messaging.py]: Is this what fixes the issue on travis? Just curious.

Comment (jamessynge) [reply pocs/tests/test_messaging.py]: multiprocessing.Event uses shared memory, while multiprocessing.Manager().Event uses proxies communicating with a shared process.

The scope=module is there just to increase the speed of the tests.

**Comment by jamessynge** [comment]
Sadly, while this worked fine on the first Travis build, it failed on the second. Need to figure out why.

File: pocs/tests/test_messaging.py (hunk 1 above)
Comment (wtgee) [reply pocs/tests/test_messaging.py]: Got, thanks.

**Comment by jamessynge** [comment]
Woohoo, latest version works both locally and in Travis.

**Pull request merged**